src/main/java/io/toast/tk/maven/plugin/RunScriptsMojo.java

```diff
@@ -0,0 +1,167 @@
+package io.toast.tk.maven.plugin;
+
+import com.google.inject.Module;
+import io.toast.tk.adapter.cache.ToastCache;
+import io.toast.tk.dao.domain.impl.test.block.ITestPage;
+import io.toast.tk.maven.plugin.run.TestPageRunner;
+import io.toast.tk.plugin.IAgentPlugin;
+import io.toast.tk.plugin.PluginLoader;
+import io.toast.tk.runtime.parse.FileHelper;
+import io.toast.tk.runtime.parse.TestParser;
+
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.model.FileSet;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.*;
+
+
+
+@Mojo(name = "run",
+        defaultPhase = LifecyclePhase.VERIFY,
+        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class RunScriptsMojo extends AbstractMojo {
+
+    @Parameter(required = false, alias = "pluginsDirectory")
+    private String pluginDir;
+
+    @Parameter(required = true, alias = "outputDirectory")
+    private String outputDirectory;
+
+    @Parameter(required = false, alias = "scripts")
+    private FileSet[] scripts;
+
+    @Parameter(required = true, defaultValue = "${project}", readonly = true)
+    private MavenProject project;
+
+    private TestPageRunner testPageRunner;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the "testPageRunner" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1450)